### ISSUE TITLE
inputOptionLabels not used for OptionComponent in ui-shared

### ISSUE THREAD

**Issue by YannickFi** [keycloak/keycloak#28078]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

account/ui

### Describe the bug

inputOptionLabels are used in the FTL but are ignored in ui-shared (account/ui).

### Version

24.0.1

### Regression

- [ ] The issue is a regression

### Expected behavior

The behaviour should match FTL templates
![image](https://github.com/keycloak/keycloak/assets/31122955/5c23d96b-3dc7-4801-a621-505faa9cdcc5)

### Actual behavior

inputOptionLabels are ignored and only the options array is used in account/ui
![image](https://github.com/keycloak/keycloak/assets/31122955/24eec411-adcf-4e07-8222-d42111825dec)

### How to Reproduce?

Create attribute with inputType multiselect-checkboxes or select-radiobuttons

```json
 {
      "name": "newsletter",
      "displayName": "${newsletter.label}",
      "validations": {
        "options": {
          "options": [
            "on"
          ]
        }
      },
      "annotations": {
        "inputOptionLabels": {
          "on": "${newsletter.text}"
        },
        "inputType": "multiselect-checkboxes"
      },
      "permissions": {
        "view": [
          "admin",
          "user"
        ],
        "edit": [
          "admin",
          "user"
        ]
      },
      "multivalued": false
    },
```

### Anything else?

https://github.com/keycloak/keycloak/blob/9caac3814c74be8f4e15dc65b0ada78ff7cb4b13/js/libs/ui-shared/src/user-profile/OptionsComponent.tsx#L7
https://github.com/keycloak/keycloak/blob/9caac3814c74be8f4e15dc65b0ada78ff7cb4b13/themes/src/main/resources/theme/base/login/user-profile-commons.ftl#L193

It has already been implemented for the Select component
https://github.com/keycloak/keycloak/blob/9caac3814c74be8f4e15dc65b0ada78ff7cb4b13/js/libs/ui-shared/src/user-profile/SelectComponent.tsx#L14

**Comment by stianst** [comment]
@edewit moving back to "Ready for backport" as this one is still marked as should be backported to 24. Please backport, or if not applicable to 24 remove the label (in which case the issue can be removed from the project completely as it doesn't need any backporting)

**Comment by edewit** [comment]
this wasn't the right fix https://github.com/keycloak/keycloak/pull/28473 fixed this and is already backported

**Comment by edewit** [comment]
This wasn't the right fix and there was another issue about this, that was fixed with #28473 that has already been backported
